test/unit/data_set_test.rb

```diff
@@ -69,6 +69,16 @@ class DataSetTest < ActiveSupport::TestCase
       assert_equal 5, PlaceArchive.all.count
     end
 
+    should "not copy ids when archiving, so that we don't get clashes" do
+      ds = @service.data_sets.first
+      PlaceArchive.create!(ds.places.first.attributes)
+      ActiveRecord::Base.connection.reset_pk_sequence!("place_archives")
```

Comment:
Note - we need this line because in the create! above we're using the id from the first place when creating the PlaceArchive, and that causes the internal id pointer to get confused. This resets it so it's sane.